backend/store/etcd/health_store.go

```diff
@@ -22,72 +25,100 @@ func isEmbeddedClient(clientURLs []string) bool {
 	return false
 }
 
+func (s *Store) getHealth(ctx context.Context, id uint64, name string, urls []string, tls *tls.Config) *corev2.ClusterHealth {
+	health := &corev2.ClusterHealth{
+		MemberID: id,
+		Name:     name,
+	}
+
+	var cli *clientv3.Client
+	var cliErr error
+
+	if isEmbeddedClient(urls) {
+		cli = s.client
+	} else {
+		cli, cliErr = clientv3.New(clientv3.Config{
+			Endpoints:   urls,
+			DialTimeout: 5 * time.Second,
+			TLS:         tls,
+		})
+	}
+
+	if cliErr != nil {
+		logger.WithField("member", id).WithField("name", name).WithError(cliErr).Error("unhealthy cluster member")
+		health.Err = cliErr.Error()
+		return health
+	}
+	defer func() {
+		_ = cli.Close()
+	}()
+
+	_, getErr := cli.Get(ctx, "health")
+
+	if getErr == nil || getErr == rpctypes.ErrPermissionDenied {
```

Comment:
Should a permission denied "swallow" errors and pretend that everything is healthy all the time?

I think it should return and explicit permission denied error so that an operator realizes they need to use the right credentials rather than think, potentially wrongly, that etcd is healthy.

Reply:
Ah, this caught my eye as well, since this logic existed before.

What I concluded was that it is correct. If etcd is returning ErrPermissionDenied, then it must be consulting its own store to do so, and therefore _is healthy_!

Reply:
I suppose if that's the existing logic, let's leave it alone...

While your logic seems to make sense, I would argue it might turn out to be a bit dangerous: "permission denied" doesn't imply "healthy". Say the definition of "healthy" involves checking that some internal metrics are within acceptable bounds. One might get a "permission denied" message, but that doesn't mean the criteria for "healthy" have been checked.